src/main/java/uk/gov/laa/ccms/caab/controller/AssessmentController.java

```diff
@@ -270,11 +282,38 @@ private void populateOpaModel(
    *         listing the answers to the questions the user has provided.
    */
   @GetMapping("/assessments/confirm")
-  public String assessmentConfirm() {
+  public String assessmentConfirm(
+      @RequestParam(value = "val") final String token,
+      final Model model
+  ) {
+    final ContextToken contextToken = contextSecurityUtil.createContextToken(token);
+    final Long rulebaseId = contextToken.getRulebaseId();
+    final String providerId = contextToken.getProviderId();
+    final String caseReferenceNumber = contextToken.getCaseId();
+
+    final AssessmentRulebase assessmentRulebase = AssessmentRulebase.findById(rulebaseId);
+
+    final AssessmentDetail assessmentDetail = Optional.ofNullable(assessmentService.getAssessments(
+            List.of(assessmentRulebase.getName()),
+            providerId,
+            caseReferenceNumber).block())
+        .map(AssessmentDetails::getContent)
+        .filter(content -> !content.isEmpty()).flatMap(content -> content.stream().findFirst())
+        .orElseThrow(() -> new CaabApplicationException("Failed to retrieve assessment details"));
+
+    //need to make sure the assessment is saved before we can display the confirmation screen
+    final List<AssessmentSummaryEntityDisplay> assessmentSummaryToDisplay =
+        assessmentService.getAssessmentSummaryToDisplay(assessmentDetail);
+
+    model.addAttribute("summary", assessmentSummaryToDisplay);
 
-    //todo - CCLS-2241 assessment confirmation screens
     return "application/assessments/assessment-confirm";
 
   }
 
+  @PostMapping("/assessments/confirm")
```

Comment:
Is this needed, if it just redirects elsewhere?

Reply:
removed